lib/checkuninitvar.cpp

```diff
@@ -131,8 +131,11 @@ void CheckUninitVar::checkScope(const Scope* scope, const std::set<std::string>
             if (tok->isStandardType() || tok->isEnumType())
                 stdtype = true;
         }
-        if (var.isArray() && !stdtype)
-            continue;
+        if (var.isArray() && !stdtype) { // std::array
+            if (!(var.isStlType() && Token::simpleMatch(var.typeStartToken(), "std :: array") && var.valueType() &&
+                  var.valueType()->containerTypeToken && var.valueType()->containerTypeToken->isStandardType()))
+                continue;
+        }
```

Comment:
This really should be fixed in ValueFlow. `checkUninitVar` has many FPs and should be removed in the future once ValueFlow can handle all the same cases. It would be best to put improvements into ValueFlow, since it will need to be there in the future either way.

Reply:
Since there is a warning for an uninitialized `char a[1]` from this function, I figured that it was the place to warn about `std::array` also.
Where would you put it in ValueFlow?

Reply:
In [valueFlowUninit](https://github.com/danmar/cppcheck/blob/main/lib/valueflow.cpp#L6746). It has some checks for `isStandardType`, that might keep it from adding the value.

Reply:
After some testing, I think valueFlowUninit() is not so useful yet.
`char a[1]`: not handled by this line:
https://github.com/danmar/cppcheck/blob/main/lib/valueflow.cpp#L6763
If I add something that passes that, I only get an `unassignedVariable` warning.
`std::array<char, 1>`: after the first few lines, `vardecl` points at `<`, If I adjust that to `a`, I hit continue at this line:
https://github.com/danmar/cppcheck/blob/main/lib/valueflow.cpp#L6772
, which makes me wonder what does valueFlowUninit() even aim at doing?

Also, what is the difference between Token::isStandardType() and ValueType::isPrimitive(), if any? Shouldn't they be named similarly, if they test the same thing?

Reply:
I misread that condition `(!var->type() && !stdtype && !pointer)`, thinking it was deliberately skipping pointers or something.
Still, `type()` is NULL, so even with special handling for `std::array` to pass the checks, the remainder of the function does not do much, and there is no warning.

Reply:
Alright, I guess its a little more complicated to support in ValueFlow. Rather than closing this issue when the PR is merged, it would be better to note it needs update to ValueFlow before closing it.